File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.cpp
class Solution {
public:
    int firstStableIndex(vector<int>& nums, int k) {
        vector<int> a;
        for (int i=1; i<=nums.size(); i++){
            a.emplace_back((*max_element(nums.begin(), nums.begin()+i))-(*min_element(nums.begin()+i-1, nums.end())));
        }
        int min_idx= -1;
        bool found= false;
        for (int i=0; i<a.size(); i++){
            if (a[i]<=k){
                if (!found){
                    min_idx=i;
                    found= true;
                }
                else if (i<min_idx){
                    min_idx=i;
                }
            }
        }
        return min_idx;
    }
};

File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.py
class Solution:
    def firstStableIndex(self, nums: list[int], k: int) -> int:
        '''instability score= max(nums[:i])-min(nums[i:len(nums)-1])
        if instability score<=k return smallest instability score else return -1'''
        a=[]
        for i in range(1, len(nums)+1):
            a.append(max(nums[:i])- min(nums[i-1:]))
        min_num= -1
        found= False
        for i in range(len(a)):
            if a[i]<=k:
                if not found:
                    min_num=i
                    found= True
                elif i<min_num:
                    min_num=i
        return min_num
                